src/components/ImageUpload.tsx

```diff
@@ -1,66 +1,130 @@
+        formData.append("upload_preset", String(PUBLIC_CLOUDINARY_UPLOAD_PRESET));
+
+        const response = await fetch(url, {
+          body: formData,
+          method: "post",
+        });
+
+        const data = await response.json();
+        setUploadedFilePublicId(data.public_id);
+        setFieldValue && setFieldValue("uploadedImageId", data.public_id, true);
+        setImgLoaded(false);
+      });
+    },
+    [setFieldValue]
+  );
+
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     accept: "image/*",
-    onDrop: acceptedFiles => {
-      setFieldValue("files", acceptedFiles);
-    },
+    multiple: false,
+    onDrop,
   });
-  return (
-    <Box>
-      { }
+  let render = <div />;
+  if (!uploadedFilePublicId) {
+    render = (
+      <div data-testid="dropzone" { ...getRootProps({ className: "dropzone" }) }>
+        <div>
+          <input
+            data-testid="drop-input"
+            { ...getInputProps() }
+            onClick={ () => {
+              setTouched && setTouched({ ["uploadedImageId"]: true });
+            } }
+          />
+          <StyledPaperInput
+            data-testid="paper-dropzone"
+            sx={ {
+              height: !isDragActive ? "38px" : "158px",
+              width: "webkit-fill-available",
+            } }
+          >
+            { isDragActive ? "Drop That Art" : dropzoneLabel }
+          </StyledPaperInput>
+          { !isDragActive ? (
+            <Box sx={ { alignItems: "center", display: "flex", flexDirection: "column", justifyContent: "center" } }>
+              <Typography variant="body1" align="center" color="primary" marginTop="30px" marginBottom="27px">
+                OR
+              </Typography>
+              <StyledFilledButton sx={ { width: "164px" } }>{ buttonLabel }</StyledFilledButton>
+            </Box>
+          ) : (
+            ""
+          ) }
+        </div>
+      </div>
+    );
+  } else {
+    render = (
       <div { ...getRootProps({ className: "dropzone" }) }>
         <input { ...getInputProps() } />
-        <StyledPaperInput
+        <Box
           sx={ {
-            width: "326px",
-            height: "38px",
+            alignItems: "center",
+            display: "flex",
+            justifyContent: "center",
+            minHeight: !imgLoaded ? "150px" : "",
           } }
         >
-          { " " }
-          Drag & Drop{ " " }
-        </StyledPaperInput>
+          { !imgLoaded ? (
+            <div data-testid="loadingIcon">
+              <CircularProgress />
+            </div>
+          ) : (
+            ""
+          ) }
 
-        <Typography variant="body1" align="center" color="primary">
-          OR
-        </Typography>
-        { isDragActive ? <p>Drop the files here ...</p> : <p>Drag n drop some files here, or click to select files</p> }
+          <Image
+            data-testid="uploadedImage"
+            style={ imgLoaded ? { display: "" } : { display: "none" } }
```

Comment:
I think that instead of changing the css display value, you could combine this with the logic from the loading indicator.
if imgLoaded is true, show the image, otherwise show the loading icon

Reply:
The idea here is that it is waiting until the image is fully loaded and rendered to show it. It needs to start rendering the image in the DOM once it gets a response from Cloudinary. The loading sign replaces the image loading into the DOM, which is why I used CSS.

Reply:
Ahh that makes sense!